[freshcabbage123] iP

freshcabbage123 opened the pull request:
# 🐍 Snake CYQJ: Conquer Your To-Do List Like Never Before

> "Order is half of life, and with Snake CYQJ, you're already halfway there."

## Why Choose Snake CYQJ? 🎯
- 🌟 **Simplicity at its Best**: Making task management as simple as a conversation.
- 🚀 **Rocket Speed**: No time wasted; Snake CYQJ is optimized for those who value speed.
- 💡 **For All Levels**: Intuitive enough for beginners, feature-rich for experts.

## Table of Contents 📝
- [ ] [Prime Features](#prime-features)
- [ ] [System Requirements](#system-requirements)
- [x] [Quick Start](#quick-start)
- [ ] [Task List](#task-list)
- [ ] [Developer Notes](#developer-notes)
- [x] [Testimonials](#testimonials)

## Prime Features 🌈🌈🌈🌈🌈🌈
- **Add Tasks**: Choose from `Todo`, `Deadline`, or `Event` to best suit your needs.
- **Status Check**: A quick `list` to see all your pending and completed tasks.
- **Seal the Deal**: `Mark` tasks as done when you complete them.
- **Task Search**: Find tasks with keywords, because memory can be a tricky thing.

## System Requirements 🖥️
- OS: Supports all major operating systems
- Java 11 or above

## Quick Start 🚀
1. [Install Java 11](https://java.com/download)
2. Download Snake CYQJ from our [Release Page].
3. Run Snake CYQJ with `java -jar duke.jar` in your terminal.

## Testimonials 🌠
> ⭐⭐⭐⭐⭐  
> "I was initially skeptical but Snake CYQJ changed my entire workflow. It's like task management for the future."  
> \- Jane D.

> ⭐⭐⭐⭐⭐  
> "Absolutely revolutionary! Snake CYQJ has made my life so much easier. It's intuitive and incredibly powerful."  
> \- Mark S.

> ⭐⭐⭐⭐  
> "Solid product. Took me a bit to get used to it, but now I can't imagine working without Snake CYQJ."  
> \- Emily R.


## Developer Notes 👨‍💻
Interested in contributing? Fork us on [GitHub](https://github.com/SnakeCYQJ).

```java
// This is just a sample code block
public class SnakeCYQJMain {
    public static void main(String[] args) {
        System.out.println("Welcome to Snake CYQJ!");
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Implement Level-0: Rename, Greet, Exit

Commit:
Implement Level-1: Echo

Commit:
Implement Level 2: Add, List

Commit:
Implement Level 3: Mark as Done

Commit:
Implement Level 4: ToDos, Events, Deadlines
Make Task class abstract.
Add subclasses of Task.

Commit:
Implement: A-TextUiTesting

Commit:
Implement Level 5: Handle Errors

Commit:
Implement: Level 6 Delete
Refactor subclasses of Task.

Commit:
Implement A-Enums
Add enums for task types.
Create TaskList class for improved OOP.

Commit:
Implement Level-7

Commit:
Merge branch-Level-7 into master

Commit:
Implement Level 8

Commit:
Merge branch 'branch-Level-8'

Commit:
Implement A-MoreOOP
Class logic is not encapsulated properly.

Extract out closely related code as classes.

Better OOP.

Commit:
Implement A-Packages
Divide code into multiple packages.

Commit:
Merge 'origin/add-gradle-support' to local branch

Commit:
Implement A-JUnit

Commit:
Implement JavaDoc changes

Commit:
Implement coding standard

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Implement Level-9

Commit:
Merge branch 'branch-Level-9'

Commit:
Implement GUI

Commit:
Merge branch 'branch-Level-10'

Respirayson commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,137 @@
+            System.out.println("An error occurred while reading the file: " + e.getMessage());
+            return new ArrayList<>();
+        }
+        ArrayList<Task> tasks = new ArrayList<>();
```
I like the naming of the variable name as plural as it is a Java Collection. Very Good 👍

Respirayson commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,45 @@
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    public String getResponse(String input) {
```
Maybe you missed it because all the other Javadoc was really good but perhaps it would be better to change this to the correct format

Respirayson commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,65 @@
+    public static String parseCommand(String input, TaskList taskList) {
+        StringBuilder output = new StringBuilder();
+        try {
+            String trimmedInput = input.trim();
```
Perhaps you could consider what if the user types in List, Bye or TODO as the command?

Respirayson commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,38 @@
+     * @param endDateTime The endDateTime of the event.
+     */
+    public Event(String description, LocalDateTime startDateTime, LocalDateTime endDateTime) {
+        super.description = description;
```
Looks a bit weird but I’m not sure whether this is an actual code style violation though 😅 It might be better to utilise the constructor. Otherwise, since this is a child class of the parent, I believe you can use this.description

Respirayson commented on src/test/java/duke/task/DeadlineTest.java:
```diff
@@ -0,0 +1,55 @@
+public class DeadlineTest {
+
+    @Test
+    public void testToString_shouldDisplayCorrectMessage() {
```
I like how you followed the naming conventions for the test names 👍

Respirayson reviewed (commented):
Looking good so far🤩🤩
Just a few nitpicks here and there

ji-just-ji reviewed (commented):
Looks like you've made good progress with your code. Maybe just do spend a bit of time looking through the style.

ji-just-ji commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,137 @@
+ * Storage class to load and save data.
+ */
+public class Storage {
+    private final Path filePath;
```
name for private final variables should be in All Caps

ji-just-ji commented on src/main/java/duke/ui/DialogBox.java:
```diff
@@ -0,0 +1,64 @@
+package duke.ui;
+
+import java.io.IOException;
```
I believe that the import statement for 'java.io.IOException' should be in the 'STANDARD_JAVA_PACKAGE' group.
Though I could be wrong.

ji-just-ji reviewed (commented):
Just a bit of style and import issues, other than that other reviewers have covered.

Commit:
Add assertions for task creation and deletion

Duke does not use assertions by default.

Assertions help to verify program state during runtime.

Enable assertions in the code at appropriate places.

This enables the easy detection of bugs.

freshcabbage123 committed:
Merge pull request #1 from freshcabbage123/branch-A-Assertions

Add assertions for task creation and deletion

Commit:
Refactor code from old text UI

Renaming of file and deleting of unused tests.

Commit:
Add functions to sort task by type and dateTime

Currently, tasks are listed in the order in which they are added.

More functionality can be added to allow users to see tasks in sorted
order.

Add 2 functions to sort task by type and dateTime in addition to the
default 'list' function.

Commit:
Merge branch 'branch-C-Sort'

Commit:
Refactor code

Create more accurate javadoc.

freshcabbage123 committed:
Merge pull request #2 from freshcabbage123/branch-A-CodeQuality

Refactor code

Commit:
Implement A-CI

Add custom workflow for Github Actions.

Commit:
docs: Implement changes based on soc-se-bot

Commit:
refactor: Refactor Parser to adhere to SLAP

Commit:
refactor: Refactor Storage based on soc-se-bot

Commit:
docs: Add javadoc for Parser

Commit:
refactor: Implement changes based on checkstyle

Commit:
fix: Resize dialog box to avoid truncation

Commit:
fix: Improve error messages for task creation

Commit:
fix: Remove dotted lines to improve UI

Commit:
test: Add and refactor existing tests

Commit:
docs: Add Ui.png

Commit:
feat: Add delay before exiting the app

Commit:
docs: Edit of README.md

Commit:
chore: Prepare for v0.2 release

Commit:
fix: Add instructions before all tests

Commit:
docs: Fix README
